plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergRegisterTableProcedure.java

```diff
@@ -0,0 +1,434 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.filesystem.FileEntry;
+import io.trino.filesystem.FileIterator;
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.hdfs.HdfsFileSystemFactory;
+import io.trino.plugin.hive.metastore.HiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.TestingConnectorSession;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static io.trino.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.plugin.iceberg.IcebergUtil.METADATA_FOLDER_NAME;
+import static io.trino.plugin.iceberg.procedure.RegisterTableProcedure.getLatestMetadataLocation;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestIcebergRegisterTableProcedure
+        extends AbstractTestQueryFramework
+{
+    private HiveMetastore metastore;
+    private File metastoreDir;
+    private TrinoFileSystem fileSystem;
+
+    public enum IcebergFileFormat
```

Comment:
In https://github.com/trinodb/trino/pull/14375#discussion_r1004365236 i meant to use the existing `IcebergFileFormat` class (`io.trino.plugin.iceberg.IcebergFileFormat`). Why do you want to have a copy?

Reply:
enum variable returns the value in the uppercase like (ORC, AVRO, and PARQUET) which was causing failure in some test cases where we access table comment and column comment. My intention was to make the format name lowercase while constructing the tableName.

Now, I have removed the local copy of `IcebergFileFormat ` and used `io.trino.plugin.iceberg.IcebergFileFormat` and then converted to lowercase wherever required.

Reply:
I could not understand, In Locally `tableName` seems to be "not case-sensitive" whereas in CI `tableName` is "case-sensitive". Test cases are failing in CI build.

Reply:
On Mac, the local filesystem is case-insensitive (by default).
On the CI it's case sensitive.